openmetrics/src/main/java/io/airlift/openmetrics/MetricsConfig.java

```diff
@@ -0,0 +1,81 @@
+package io.airlift.openmetrics;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+
+import java.util.List;
+
+public class MetricsConfig
+{
+    private boolean sanitizeMetricNames;
+    private boolean inferNumeric = true;
+    private boolean inferStrings;
+    private boolean inferAll;
+    private List<String> jmxDomains = ImmutableList.of();
+
+    public boolean isSanitizeMetricNames()
+    {
+        return sanitizeMetricNames;
+    }
+
+    @Config("openmetrics.sanitize-metric-names")
+    @ConfigDescription("Prevent metric names that don't match the openmetrics format to be returned from the metrics endpoint. This can be costly in CPU time if metrics are frequently scraped.")
```

Comment:
Why would we allow unsanitary metric names at all

Reply:
There shouldn't be any unsanitary names, but this enables an extra check that will add some additional logging if the fetching agent (datadog, prometheus, etc.) doesn't like the names we generate.

Reply:
Removed with the precompiled regex and the memoization, it'll be fast enough now..